modules/katamari/src/main/ts/ephox/katamari/api/Type.ts

```diff
@@ -1,35 +1,43 @@
-const typeOf = function (x: any) {
-  if (x === null) {
-    return 'null';
-  }
-  const t = typeof x;
-  if (t === 'object' && (Array.prototype.isPrototypeOf(x) || x.constructor && x.constructor.name === 'Array')) {
-    return 'array';
-  }
-  if (t === 'object' && (String.prototype.isPrototypeOf(x) || x.constructor && x.constructor.name === 'String')) {
-    return 'string';
-  }
-  return t;
-};
+const typeOf = (x: any): string =>
+  x === null ? 'null' : isArray(x) ? 'array' : isString(x) ? 'string' : typeof x;
 
-const isType = function (type: string) {
-  return function (value: any) {
-    return typeOf(value) === type;
-  };
-};
+const isType = <Yolo>(type: string) => (value: any): value is Yolo =>
+  typeOf(value) === type;
+
+const isSimpleType = <Yolo>(type: string) => (value: any): value is Yolo =>
+  typeof value === type;
+
+const eq = <T> (t: T) => (a: any): a is T =>
+  t === a;
+
+export const isString = (x: any): x is string =>
+  x !== null && typeof x === 'object' && (String.prototype.isPrototypeOf(x) || x.constructor && x.constructor.name === 'String');
+
+export const isObject: (value: any) => boolean =
+  isType('object');
+
+export const isArray = (value: any): value is Array<unknown> =>
```

Comment:
This likely will not work across different frames, as that's why we had the `constructor.name` thing.

Reply:
I wonder if that's why the build broke...

Reply:
Nope. Build node out of space.

Reply:
Sorry, I didn't mean to say not to use `Array.isArray`,  just that we'd likely still need the `x.constructor && x.constructor.name === 'Array'` part to handle cross frame checks.

Reply:
Hmmm